### PR TITLE
add option to limit test run time time

### PR DESCRIPTION
**Pull request by bernt-matthias**
fixes https://github.com/galaxyproject/planemo/issues/999

not sure if we need an option in planemo / if setting the env variable in planemo ci action would be sufficient (but a bit more complex in bash, since it must be checked against zero)

### CODE REVIEW

File: planemo/options.py [hunk 1]
```diff
@@ -1591,6 +1591,12 @@ def test_options():
                 "to disable completely."
             ),
         ),
+        planemo_option(
+            "--test_default_wait",
```
Comment (mvdbeek) [inline planemo/options.py:1595]: How about naming the option `--test_timeout` ?

**Review by mvdbeek — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Seems that this is ignored if not run with `run_tests.sh` .. would be good to have a test to see it breaking in the future :) But no idea how and where.

**Comment by mvdbeek** [comment]
Yes, a test would be great. You can add a tool that sleeps for a long time and add a test to `test_cmd_test.py` at the end you can make sure that the test finished faster than the sleep you're using in the tool.

**Comment by bernt-matthias** [comment]
OK. Test added. Seems that currently none of the tests run if `@skip_if_environ("PLANEMO_SKIP_GALAXY_TESTS")` is used... Maybe I'm wrong. If not, I guess this is a bug in CI, or?

This will be my last message before vacation. Feel free to modify this or wait for my return mid June .. Cheers.

File: tests/test_cmd_test.py [hunk 2]
```diff
@@ -53,6 +54,31 @@ def test_workflow_test_simple_yaml(self):
...
+                test_artifact,
+            ]
+            self._check_exit_code(test_command, exit_code=1)
+            with open(json_out.name) as fh:
+                tool_test_json = json.load(fh)
+                assert tool_test_json["summary"]["num_tests"] == 1
+                # check run time, for smaller 10 since the test will take a bit longer than 1s
+                # the important bit is that it's not about 30s (since the test tool calls `sleep 30`)
+                assert (
+                    float(tool_test_json["tests"][0]["data"]["time_seconds"]) > 10
```
Comment (nsoranzo) [inline tests/test_cmd_test.py:76]:
```suggestion
                    float(tool_test_json["tests"][0]["data"]["time_seconds"]) <= 10
```

File: tests/test_cmd_test.py [hunk 3]
```diff
@@ -53,6 +54,31 @@ def test_workflow_test_simple_yaml(self):
...
+                ), "Test needed more than 10 sec but should time out after 1"
+                assert (
+                    "Timed out after" not in tool_test_json["tests"][0]["data"]["problem_log"]
```
Comment (nsoranzo) [inline tests/test_cmd_test.py:79]:
```suggestion
                    "Timed out after" in tool_test_json["tests"][0]["data"]["problem_log"]
```

**Comment by nsoranzo** [comment]
@mvdbeek The test passes for me with the changes in my review. If you agree they are correct, I think we can apply them and merge the PR without waiting for Matthias to come back.

**Pull request merged**